### PR TITLE
[Blosc] add recipe

### PR DESCRIPTION
**Pull request by jaakkor2**
Compared to old builder https://github.com/stevengj/BloscBuilder/blob/master/build_tarballs.jl
* Updates version from 0.14.3 to 0.18.1
* Disables static lib
* Uses external Zlib, Zstd, Lz4.  Snappy commented out, could be uncommented once available.

### CODE REVIEW

**Comment by jaakkor2** [comment]
Dies with
`x86_64-w64-mingw32-gcc: error: unrecognized command line option ‘-rdynamic’`

File: B/Blosc/build_tarballs.jl [hunk 1]
```diff
@@ -24,6 +24,14 @@ CMAKE_FLAGS+=(-DPREFER_EXTERNAL_LZ4=ON)
...
 make -j${nproc}
 make install
+
+install_license ${WORKSPACE}/srcdir/LICENSES/BITSHUFFLE.txt
+install_license ${WORKSPACE}/srcdir/LICENSES/BLOSC.txt
+install_license ${WORKSPACE}/srcdir/LICENSES/FASTLZ.txt
+install_license ${WORKSPACE}/srcdir/LICENSES/LZ4.txt
+install_license ${WORKSPACE}/srcdir/LICENSES/SNAPPY.txt
+install_license ${WORKSPACE}/srcdir/LICENSES/STDINT.txt
+install_license ${WORKSPACE}/srcdir/LICENSES/ZLIB.txt
```
Comment (giordano) [inline B/Blosc/build_tarballs.jl:34]:
```suggestion
install_license ../LICENSES/*.txt
```
would do the job as well :wink:

**Comment by ViralBShah** [comment]
Wow I have never seen something build only on freebsd and nowhere else. Usually it's the opposite. :-)

**Comment by giordano** [comment]
The problem was that the the cmake weren't actually passed to the `cmake` command: `CMAKE_FLAGS` is a vector and you need to use `${CMAKE_FLAGS[@]}` to expand all of its elements, not just the first one.

Now we're [getting an interesting error](https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=2591&view=logs&jobId=0edecaae-5d09-5904-7944-ef5c775233bb&j=dd28c299-4f9d-572e-69f9-aa6d8572b131&t=4748de3d-0218-5c73-d1ac-e62def639361) for `i686-w64-mingw32`:
```
[03:34:32] [100%] Linking CXX shared library libblosc.dll
[03:34:32] cd /workspace/srcdir/c-blosc-1.18.1/build/blosc && /usr/bin/cmake -E cmake_link_script CMakeFiles/blosc_shared.dir/link.txt --verbose=true
[03:34:32] /usr/bin/cmake -E remove -f CMakeFiles/blosc_shared.dir/objects.a
[03:34:32] /opt/i686-w64-mingw32/bin/i686-w64-mingw32-ar cr CMakeFiles/blosc_shared.dir/objects.a @CMakeFiles/blosc_shared.dir/objects1.rsp
[03:34:32] /opt/bin/i686-w64-mingw32-g++ --sysroot=/opt/i686-w64-mingw32/i686-w64-mingw32/sys-root/  -O3 -DNDEBUG  -shared -o libblosc.dll -Wl,--out-implib,libblosc.dll.a -Wl,--major-image-version,1,--minor-image-version,18 -Wl,--whole-archive CMakeFiles/blosc_shared.dir/objects.a -Wl,--no-whole-archive @CMakeFiles/blosc_shared.dir/linklibs.rsp
[03:34:32] CMakeFiles/blosc_shared.dir/objects.a(shuffle.c.obj):shuffle.c:(.text+0x3e3): undefined reference to `InitOnceBeginInitialize'
[03:34:32] CMakeFiles/blosc_shared.dir/objects.a(shuffle.c.obj):shuffle.c:(.text+0x43d): undefined reference to `InitOnceComplete'
[03:34:32] CMakeFiles/blosc_shared.dir/objects.a(shuffle.c.obj):shuffle.c:(.text+0x473): undefined reference to `InitOnceBeginInitialize'
[03:34:32] CMakeFiles/blosc_shared.dir/objects.a(shuffle.c.obj):shuffle.c:(.text+0x4cd): undefined reference to `InitOnceComplete'
[03:34:32] CMakeFiles/blosc_shared.dir/objects.a(shuffle.c.obj):shuffle.c:(.text+0x517): undefined reference to `InitOnceBeginInitialize'
[03:34:32] CMakeFiles/blosc_shared.dir/objects.a(shuffle.c.obj):shuffle.c:(.text+0x5bd): undefined reference to `InitOnceComplete'
[03:34:32] CMakeFiles/blosc_shared.dir/objects.a(shuffle.c.obj):shuffle.c:(.text+0x607): undefined reference to `InitOnceBeginInitialize'
[03:34:32] CMakeFiles/blosc_shared.dir/objects.a(shuffle.c.obj):shuffle.c:(.text+0x6ad): undefined reference to `InitOnceComplete'
[03:34:32] collect2: error: ld returned 1 exit status
[03:34:32] make[2]: *** [blosc/CMakeFiles/blosc_shared.dir/build.make:298: blosc/libblosc.dll] Error 1
[03:34:32] make[2]: Leaving directory '/workspace/srcdir/c-blosc-1.18.1/build'
[03:34:32] make[1]: *** [CMakeFiles/Makefile2:123: blosc/CMakeFiles/blosc_shared.dir/all] Error 2
[03:34:32] make[1]: Leaving directory '/workspace/srcdir/c-blosc-1.18.1/build'
[03:34:32] make: *** [Makefile:152: all] Error 2
```

At a quick glance, I don't see why this should fail only for this platform and not for `x86_64-w64-mingw32` as well :confused:

**Comment by giordano** [comment]
I've changed the version to 1.14.3, the same used in https://github.com/stevengj/BloscBuilder.  There seem to be problems with `i686-w64-mingw32` in newer versions, I'd suggest to use this one that is known to work with all supported platforms.  @jaakkor2 are you ok with this?

**Comment by jruohio-invn** [comment]
> 
> 
> I've changed the version to 1.14.3, the same used in https://github.com/stevengj/BloscBuilder. There seem to be problems with `i686-w64-mingw32` in newer versions, I'd suggest to use this one that is known to work with all supported platforms. @jaakkor2 are you ok with this?

Sure!  Thanks for fixing most of the problems.

**Comment by jaakkor2** [comment]
> 
> 
> I've changed the version to 1.14.3, the same used in https://github.com/stevengj/BloscBuilder. There seem to be problems with `i686-w64-mingw32` in newer versions, I'd suggest to use this one that is known to work with all supported platforms. @jaakkor2 are you ok with this?

Sure!  Thanks for fixing most of the problems.

**Pull request merged**